sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -63,11 +65,15 @@ abstract class DataSourceV2SQLSuite
 
   protected def assertAnalysisError(
       sqlStatement: String,
-      expectedError: String): Unit = {
-    val ex = intercept[AnalysisException] {
-      sql(sqlStatement)
-    }
-    assert(ex.getMessage.contains(expectedError))
+      errorClass: String,
+      parameters: Map[String, String]): Unit = {
+    checkError(
+      exception = intercept[AnalysisException] {
+        sql(sqlStatement)
+      },
+      errorClass = errorClass,
+      parameters = parameters
+    )
```

Comment:
Let's introduce a helper method for this:
```scala
      exception = intercept[AnalysisException] {
        sql(sqlStatement)
      },
```
and invoke `checkError` directly.

Reply:
Done